pkg/controller/cache/cluster/controller.go

```diff
@@ -147,10 +150,26 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 		return managed.ExternalObservation{}, err
 	}
 
-	return managed.ExternalObservation{
+	obs := managed.ExternalObservation{
 		ResourceExists:   true,
 		ResourceUpToDate: upToDate,
-	}, nil
+	}
+
+	if cr.Status.AtProvider.CacheNodes != nil {
+		nodes := []string{}
+		for _, n := range cr.Status.AtProvider.CacheNodes {
+			addr := fmt.Sprintf("%s:%d", n.Endpoint.Address, n.Endpoint.Port)
+			nodes = append(nodes, addr)
+		}
+		
+		obs.ConnectionDetails = managed.ConnectionDetails{
+			xpv1.ResourceCredentialsSecretEndpointKey: []byte(
+				strings.Join(nodes, ","),
+			),
+		}
```

Comment:
I think we should keep a single "default" endpoint and and keep the other endpoints in seperate keys, maybe `endpoint2`, `endpoint3` or add a second key `endpoints` with the joined endpoints string.